src/systems/filecoin_markets/deal/_index.md

```diff
@@ -3,6 +3,52 @@ menuTitle: Deals
+    - a. `StorageProvider` calls `PublishStorageDeals` in `StorageMarketActor` which will generate a `DealID` for each `StorageDeal` and store a mapping from `DealID` to `StorageDeal`. However, the deals are not active at this point.
+      - As a backup, `StorageClient` MAY call `PublishStorageDeals` with the `StorageDeal`, to activate the deal.
+      - It is possible for either `StorageProvider` or `StorageClient` to try to enter into two deals simultaneously with funds available only for one. Only the first deal to commit in the chain would clear, the second would fail with error `errorcode.InsufficientFunds`.
+    - b. `StorageProvider` calls `HandleStorageDeal` in `StorageMiningSubsystem` which will then add the `StorageDeal` into a `Sector`.
+- 4. Once the miner finishes packing a `Sector`, it generates a Sealed Sector and calls `StorageMinerActor.CommitSector` to verify the seal, store sector expiration, and record the mapping from `SectorNumber` to `SealCommitment`. It will also place this newly added `Sector` in the list of `CommittedSectors` in `StorageMinerActor`. `StorageMiner` does not earn any power for this newly added sector until its first PoSt has been submitted.
+
+# Declare and Recover Faults
+
+- 5. Declared faults are penalized to a smaller degree than spotted faults by `CronActor`. Miners declare faulty sectors by invoking `StorageMinerActor.DeclareFaults` and X of the `StorageDealCollateral` will be slashed and power corresponding to these sectors will be tempororily lost.
+- 6. Miners can then recover faults by invoking `StorageMinerActor.RecoverFaults` and have sufficient `StorageDealCollateral` in their available balances. FaultySectors are recommitted and power is only restored at the next PoSt submission.
+- 7. Sectors that are declared faulty for `storagemining.MaxFaults` consecutive ChainEpochs will result in `StoragePowerActor.SlashPledgeCollateral`.
+  - TODO: set `X` parameter
+
+# Submit PoSt
+
+(TODO: move into Storage Mining)
+
+On every PoSt Submission, the following steps happen.
+
+- 8. `StorageMinerActor` first verifies the PoSt Submission. All Sectors will be considered in `SpottedFaults` if PoSt submission has failed (move to State 14).
+- 9. If `CommittedSectors` are proven in `PoStSubmission.SectorSet`, Storage Miner gains power for these newly committed sectors.
+- 10. If there are `DeclaredFaultySectors` , `Sector` in that set will not be challenged.
+- 11. For all other sectors, payment will be processed by invoking `StorageMarketActor.ProcessStorageDealsPayment` and miner available balances will be updated.
+- 12. Decide which Sectors have expired by looking at the `SectorExpirationQueue`. Sectors expire when all deals in that Sector have expired. `StorageDealCollateral` for both miners and users will only be returned when all deals in the Sector have expired. This is done by calling `StorageMarketActor.SettleExpiredDeals` and the Sector will be deleted from `StorageMinerActor.Sectors`.
```

Comment:
> Sectors expire when all deals in that Sector have expired.

This is an unexpected change to me. IMO miners should explicitly remove sectors when they no longer want to claim power for them, not have them automatically lose power. Collateral can remain pinned until that explicit action if necessary. I don't think we need a concept of sector expiration at all, unless it refers to the explicit action taken by a miner (but then "done" would be a better word).

If there's an explicit protocol decision to have storage power expire automatically, that should be called out loudly. It implies a lot more re-sealing work for miners who want to maintain power but don't have new, paying client deals in enough quantity to fill their storage. And a difficult decision when self-dealing to determine how long to set the deal before paying the cost of re-sealing. One might hope for *short* self-deals so that storage can be quickly repurposed to client deals, but this mechanism incentivises very long self deals, which might exclude unknown future client deals. @jbenet